Спринт №9

Shep0rt opened the pull request:
- Создал общий стиль для атрибутов на главном экране и в настройках
- Убрал тени с кнопок
- Сделал адаптацию экранов под темный режим
- Реализовал кнопки на экране "Настройки": «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение»
- Реализовал экран "Поиск" и функционал поля поиска на нем
- Реализовал сохранение и восстановление данных в поле поиска

Shep0rt committed:
Спринт №9
- Создал общий стиль для атрибутов на главном экране и в настройках
- Убрал тени с кнопок
- Сделал адаптацию экранов под темный режим
- Реализовал кнопки на экране "Настройки": «Поделиться приложением», «Написать разработчикам», «Пользовательское соглашение»
- Реализовал экран "Поиск" и функционал поля поиска на нем
- Реализовал сохранение и восстановление данных в поле поиска

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,11 +1,91 @@
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                clearButton.visibility = clearButtonVisibility(s)
```
Рекомендация:
Можно сделать чуть проще: `clearButton.isVisible = !s.isNullOrEmpty()`
А метод `clearButtonVisibility`, соответственно, удалить

Shep0rt committed:
Исправления по спринту №9
- Исправил цвет курсора в поле "Поиск" на корректный
- Упростил механизм отображения кнопки очистки поля "Поиск"
- Исправил работу кнопки очистки поля в поле "Поиск"
